backend/src/packages/users/user.service.ts

```diff
@@ -51,6 +52,21 @@ class UserService implements IService {
     };
   }
 
+  public async checkUserPassword(
+    payload: UserSignInRequestDto,
+  ): Promise<boolean> {
+    const { email, password } = payload;
+
+    const user = await this.userRepository.findByEmail(email);
+
+    return user
+      ? await this.encrypt.checkPassword(
+          password,
+          user.toNewObject().passwordHash,
+        )
+      : false;
+  }
```

Comment:
And create new static getter `privateData` upon user entity that returns `passwordHash` and `passwordSalt`
```suggestion
  public async findPrivateData(id: number): Promise<UserPrivateData | null> {
    const user = await this.userRepository.find(id);

    if (!user) {
      return null;
    }

    return user.privateData;
  }
```